### DISCUSSION TITLE
usage on intel nuc6ayh cpu (J3455) & hw accel

### DISCUSSION THREAD

**Discussion by LeifSbr** [blakeblackshear/frigate#4575] [Ask A Question] [answered]
hello chaps & chapettes...
my home assistant supervised machine seems to be crashing lately. one thing i've noticed cpu usage has seemingly shot up, not sure if it  was a home assistant update... i have two corals on usb and they seem to be detecting fine (eight camera streams @ 5fps/1280x720).
interestingly top is reporting huge difference to the home assistant cpu usage sensor (50-55% vs 85-90%)... i'm using intel < 10th gen hw accel settings as per the docs:

```
ffmpeg:
  hwaccel_args:
    - -hwaccel
    - vaapi
    - -hwaccel_device
    - /dev/dri/renderD128
    - -hwaccel_output_format
    - yuv420p
```

interestingly the intel >= 10th gen  settings *seem* to work as well (?), albeit intel-gpu-top is reporting different utilization characteristics...  "Render/3D/0" is not utilized at all but its reading much higher "Video/0" usage.

i'm no video/codec/streaming anorak so am not quite sure if i'm using the correct arguments? any input/advice would be great :)

![2022-12-02 14_44_06-Window](https://user-images.githubusercontent.com/28689427/205298093-9876fb7a-b6cc-489c-8c14-9bd62dc814f0.png)

when using:
```
ffmpeg:
  hwaccel_args: -c:v h264_qsv

```
![2022-12-02 15_09_11-Window](https://user-images.githubusercontent.com/28689427/205300208-5d368e92-c632-42d6-a87d-bb3bb2100467.png)

**Comment by NickM-27** [comment, marked as answer]
A few things here:
1. `top` reports CPU usage per cpu core meanwhile HA just reports general usage. 
2. Your processes using the most CPU resources aren't ffmpeg meaning it's not the decoding that is using your CPU. Looks like you have a lot of motion going on

**Reply by LeifSbr** [reply]
> 1\. top reports CPU usage per cpu core meanwhile HA just reports general usage.

thank you! now it makes sense...

> 2\. Looks like you have a lot of motion going on

...come to think of it, past few days have been pretty windy. i do use masks (to block out foliage mostly) and only detect for persons & cars.

...i also happened to read your other reply on another [thread](https://github.com/blakeblackshear/frigate/discussions/3506):

"In 0.11+ we have added an option to toggle motion detection. This is just an additional toggle that will reduce CPU a bit further but not nearly as much as the detection switch (in CPU detector case)"

yup this would quite useful... though i just looked at my version... 0.10.1. time to upgrade it seems. thanks again.

**Reply by NickM-27** [reply]
You might have some issues with hwaccel on upgrade so definitely recommend reading the 0.11 release notes.

**Reply by blakeblackshear** [reply]
Note that turning off motion detection also requires object detection to be turned off. It won't store anything unless you have 24/7 recording on in all mode.

**Reply by LeifSbr** [reply]
> You might have some issues with hwaccel on upgrade so definitely recommend reading the 0.11 release notes.

much appreciated. i added the "LIBVA_DRIVER_NAME=i965" environment variable, it seems to yield the best results. process calmed downed a wee bit (but then, so has the wind, lol).

![2022-12-02 17_38_10-Window](https://user-images.githubusercontent.com/28689427/205330927-2726ea66-9e38-4d0b-9c0e-52c25ca1e25c.png)
